src/protocolsupportpocketstuff/util/EntitySpawnListener.java

```diff
@@ -0,0 +1,119 @@
+	private Connection con;
+
+	public EntitySpawnListener(ProtocolSupportPocketStuff plugin, Connection con) {
+		this.plugin = plugin;
+		this.con = con;
+	}
+
+	@Override
+	public void onRawPacketSending(RawPacketEvent event) {
+		super.onRawPacketSending(event);
+
+		ByteBuf data = event.getData();
+
+		int packetId = VarNumberSerializer.readVarInt(data);
+
+		if (packetId != PEPacketIDs.SPAWN_PLAYER && packetId != PEPacketIDs.SPAWN_ENTITY)
+			return;
+
+		data.readByte();
+		data.readByte();
+
+		UUID uuid = null;
+		String name = null;
+
+		if (packetId == PEPacketIDs.SPAWN_PLAYER) {
+			// UUID and stuff... we don't care about this but we need to read them anyway
+			uuid = MiscSerializer.readUUID(data);
+			name = StringSerializer.readString(data, ProtocolVersion.MINECRAFT_PE);
+		}
+
+		long entityId = VarNumberSerializer.readSVarLong(data);
+
+		if (!PocketUtils.hasCustomScale((int) entityId)) // Does this entity has custom scale?
+			return; // No? Then whatever, I'm outta here!
+
+		ByteBuf serializer = Unpooled.buffer(); // Let's start rewriting our own:tm: packet
+		VarNumberSerializer.writeVarInt(serializer, packetId);
+		serializer.writeByte(0);
+		serializer.writeByte(0);
+		if (packetId == PEPacketIDs.SPAWN_PLAYER) {
+			MiscSerializer.writeUUID(serializer, uuid);
+			StringSerializer.writeString(serializer, ProtocolVersion.MINECRAFT_PE, name);
+		}
+		VarNumberSerializer.writeSVarLong(serializer, (long) entityId);
+		VarNumberSerializer.writeVarLong(serializer, VarNumberSerializer.readVarLong(data)); // runtime ID
+
+		if (packetId == PEPacketIDs.SPAWN_ENTITY) {
+			VarNumberSerializer.writeVarInt(serializer, VarNumberSerializer.readVarInt(data)); // type
+		}
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // position
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // position
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // position
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // motion
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // motion
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // motion
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // pitch
+		MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // yaw (or head yaw)
+		if (packetId == PEPacketIDs.SPAWN_PLAYER) {
+			MiscSerializer.writeLFloat(serializer, MiscSerializer.readLFloat(serializer)); // yaw
+			VarNumberSerializer.writeSVarInt(serializer, VarNumberSerializer.readSVarInt(serializer)); // slot
+		}
+		// now comes the cool stuff:tm:
+		int entries = VarNumberSerializer.readVarInt(serializer);
+		VarNumberSerializer.writeVarInt(serializer, entries); // current metadata + scale
+
+		HashMap<Integer, DataWatcherObject> dataWatchers =  new HashMap<Integer, DataWatcherObject>();
```

Comment:
This is the perfect place for an array with set size. Btw, I believe you can get these things from within PS itself.